sort by translation date if it exists, otherwise use pub date

mdlincoln opened the pull request:
Closes #1251

This adjusts the code that creates the "sort date" when rendering a lesson page. IF a lesson has a translated date, that's the sorting date. If it is an original lesson without a translation date, then the sort date will be the publication date.

@programminghistorian/spanish-team needs to update the `sort-by-date` entry in `_snippets.yml`.

mdlincoln committed:
sort by translation date if it exists, otherwise use pub date

mdlincoln committed:
more accurate comment

mdlincoln committed:
validate that translation dates > original dates

mdlincoln commented:
While I'm at it, I've also added a check that `translation_date` is greater than `date`

jenniferisasi reviewed (dismissed):
Great! And thanks for the comments you enter on the code itself, is very helpful!

jenniferisasi committed:
Update snippets.yml

Per #1251 and fix #1252 I have updated the snippet text for the "sort-by" of lessons, from date of translation to date of publication as we now have both types of lesson.

jenniferisasi commented:
Approved @mdlincoln's changes to sort publications and changed snippet to reflect original publications in spanish.
